### PR TITLE
Verifier for scriptable changes

### PR DESCRIPTION
**Pull request by str4d**
Includes changes from the following upstream PRs:
- bitcoin/bitcoin#10189
  - Excluding the `CNode` scripted changes.
- bitcoin/bitcoin#10480
- bitcoin/bitcoin#11390
- bitcoin/bitcoin#13281
  - Only the lint scripts we already have.
- bitcoin/bitcoin#13454
  - Only changes to scripts we already have.
- bitcoin/bitcoin#14864
- bitcoin/bitcoin#16327
  - Includes some portability fixes to other shell scripts.
- bitcoin/bitcoin#20069

### CODE REVIEW

File: contrib/qos/tc.sh [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
```
Comment (daira) [inline contrib/qos/tc.sh:3]:
```suggestion
# file COPYING or https://www.opensource.org/licenses/mit-license.php .
```

File: doc/developer-notes.md [hunk 2]
```diff
@@ -243,4 +243,25 @@ ACK -  A loose ACK can be confusing. It's best to avoid them unless it's a docum
...
+
+- start the commit message with `scripted-diff:` (and then a description of the diff on the same line)
+- in the commit message include the bash script between lines containing just the following text:
+
+    - `-BEGIN VERIFY SCRIPT-`
+    - `-END VERIFY SCRIPT-`
+
+The scripted-diff is verified by the tool `test/lint/commit-script-check.sh`
+
+Commit `bb81e173` is an example of a scripted-diff.
```
Comment (str4d) [inline doc/developer-notes.md:267]: This commit only exists upstream. We can use the one commit in this PR that makes a scripted diff.
```suggestion
Commit `ccd074a5` is an example of a scripted-diff.
```

File: test/lint/commit-script-check.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
```
Comment (daira) [inline test/lint/commit-script-check.sh:4]:
```suggestion
# file COPYING or https://www.opensource.org/licenses/mit-license.php .
```

**Review by daira — APPROVED** [review]
utACK

**Comment by daira** [comment]
ut-reACK 8aadc745c25f0af2f49a6fbeb35610f44432a636

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**